Packs/Iboss/Integrations/Iboss/Iboss.py

```diff
@@ -0,0 +1,1329 @@
+            params={"currentPolicyBeingEdited": current_policy_being_edited},
+            error_handler=self._list_error_handler,
+            ok_codes=(200, 201, 202, 204, 422)
+        )
+        return result
+
+    @set_auth
+    def remove_entity_from_block_list(self, url, current_policy_being_edited=1, apply_global=0):
+        params = {
+            "url": url,
+            "currentPolicyBeingEdited": current_policy_being_edited,
+            "global": apply_global,
+        }
+
+        result = self._http_request(
+            method="delete",
+            url_suffix="/json/controls/blockList",
+            params=params,
+            ok_codes=(200, 201, 202, 204, 422),
+            error_handler=self._remove_from_list_error_handler,
+        )
+        return result
+
+    @set_auth
+    def add_entity_to_allow_list(self, url, current_policy_being_edited=1, allow_keyword=0,
+                                 direction=2, end_port=0, apply_global=0, is_regex=0, is_timed_url=0, note="",
+                                 priority=0, start_port=0, time_url_expires_in_minutes=0):
+        params = {
+            "url": url,
+            "allow_keyword": allow_keyword,
+            "direction": direction,
+            "endPort": end_port,
+            "global": apply_global,
+            "isRegex": is_regex,
+            "isTimedUrl": is_timed_url,
+            "note": note,
+            "priority": priority,
+            "startPort": start_port,
+            "timedUrlExpiresInMinutes": time_url_expires_in_minutes,
+        }
+
+        result = self._http_request(
+            method="put",
+            url_suffix="/json/controls/allowList",
+            json_data=params,
+            params={"currentPolicyBeingEdited": current_policy_being_edited},
+            error_handler=self._list_error_handler,
+            ok_codes=(200, 201, 202, 204, 422)
+        )
+        return result
+
+    @set_auth
+    def remove_entity_from_allow_list(self, url, current_policy_being_edited=1, apply_global=0):
+        params = {
+            "url": url,
+            "currentPolicyBeingEdited": current_policy_being_edited,
+            "global": apply_global,
+        }
+
+        result = self._http_request(
+            method="delete",
+            url_suffix="/json/controls/allowList",
+            params=params,
+            ok_codes=(200, 201, 202, 204, 422),
+            error_handler=self._remove_from_list_error_handler
+        )
+        return result
+
+    @set_auth
+    def get_url_reputation(self, url):
+        data = {"url": url}
+        result = self._http_request(
+            method="post",
+            url_suffix="/json/controls/urlLookup",
+            json_data=data
+        )
+        return result
+    # TODO: ADD HERE THE FUNCTIONS TO INTERACT WITH YOUR PRODUCT API
+
+
+''' HELPER FUNCTIONS '''
+
+def _iboss_entity_lookup(client, entity, _type, sub_type):
+    response = client.get_url_reputation(entity)
```

Comment:
This is a result of the naming schema for that method in iboss. The actual API endpoint is the URL lookup, but the URL lookup works with IPs and domains (in those cases they will likely be redirected to the URL-version of an IP or domain when the checks are made in iboss). This is why I named the customer-facing command in Demisto `entity` lookup, since that implies it can be used for more types than only URL.

Would it be best if I renamed the actual client method to `get_entity_reputation`?

Reply:
Added docstring commentary